lib/front_matter_parser/writer.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require_relative "parser"
+require_relative "dumper"
+require_relative "syntax_writer"
+
+module FrontMatterParser
+  # Entry point to parse a front matter from a string or file.
+  class Writer < Parser
+    # @!attribute [r] dumper
+    # Current dumper in use. See {Dumper} for details
+    attr_reader :dumper
+    attr_reader :syntax_writer
+
+    class << self
+      def write_file(pathname, parsed, syntax_parser: nil, **options)
```

Comment:
Instead of  `**options`, we could be explicit and use `dumper:`, just to be consistent with the parser sibling.